File: android/app/src/main/kotlin/com/vitoksmile/cpmoviemaker/provider/FFmpegProviderImpl.kt
package com.vitoksmile.cpmoviemaker.provider

import com.arthenica.mobileffmpeg.Config
import com.arthenica.mobileffmpeg.FFmpeg

class FFmpegProviderImpl : FFmpegProvider {
    override val returnCodeSuccess get() = Config.RETURN_CODE_SUCCESS
    override val returnCodeCancel get() = Config.RETURN_CODE_CANCEL

    override fun execute(arguments: List<String>): Int {
        return FFmpeg.execute(arguments.toTypedArray())
    }

    override fun cancel() {
        FFmpeg.cancel()
    }
}

File: ios/Runner/FFmpegProviderImpl.swift
import Foundation
import SharedCode
import mobileffmpeg

class FFmpegProviderImpl : FFmpegProvider {
    let returnCodeSuccess: Int32 = RETURN_CODE_SUCCESS
    let returnCodeCancel: Int32 = RETURN_CODE_CANCEL
    
    func execute(arguments: [String]) -> Int32 {
        return MobileFFmpeg.execute(withArguments: arguments)
    }
    
    func cancel() {
        MobileFFmpeg.cancel()
    }
}
